packages/salesforcedx-vscode-lwc/src/metasupport/metaSupport.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as fs from 'fs';
+import * as vscode from 'vscode';
+import path = require('path');
+import semver = require('semver');
+
+const EXTENSION_NAME = 'salesforce.salesforcedx-vscode-lwc';
+
+export class MetaSupport {
+  private static instance: MetaSupport;
+  private static sfdxUri = '.sfdx';
+  private static resourceUri = path.join(MetaSupport.sfdxUri, 'resources');
+  private static lwcResourceUri = path.join(
+    MetaSupport.resourceUri,
+    'lwcResources'
+  );
+  private static dir = path.join(
+    vscode.workspace.rootPath!,
+    MetaSupport.lwcResourceUri
+  );
+  private static resourceDir = path.join(
+    vscode.workspace.rootPath!,
+    MetaSupport.resourceUri
+  );
+
+  public static initializeSupport() {
+    if (!MetaSupport.instance) {
+      MetaSupport.instance = new MetaSupport();
+    }
+    return MetaSupport.instance;
+  }
+
+  /**
+   * Creates LWCResource folder if not exist
+   */
+  private createLWCResourceFolder() {
+    if (!fs.existsSync(MetaSupport.resourceDir)) {
+      fs.mkdirSync(MetaSupport.resourceDir);
+      fs.mkdirSync(MetaSupport.dir);
+    } else if (!fs.existsSync(MetaSupport.dir)) {
+      fs.mkdirSync(MetaSupport.dir);
+    }
+  }
+
+  /**
+   * Copies static XSD and XML files to .sfdx folder.
+   * TODO: use npm install to deliever these files.
+   */
+  private getLocalFile(targetFileName: string, destinationPath: string) {
+    const thisExtPath = vscode.extensions.getExtension(
+      'salesforce.salesforcedx-vscode-lwc'
+    )!.extensionPath;
+    const resourcepath = path.join(
+      thisExtPath,
+      'resources',
+      'static',
+      targetFileName
+    );
+    fs.copyFileSync(resourcepath, destinationPath);
```

Comment:
Why does it have to copy them? Can't you just pass the `thisExtPath` into the redhat extension?